### DISCUSSION TITLE
Suggestion for template helper: first select device class, filter Unit of measurement dropdown

### DISCUSSION THREAD

**Discussion by balk77** [home-assistant/frontend#17863] [Dashboard]
I really like the new template helper. One suggestion: put the Device Class selector above the unit of measurement selector. And when a Device Class is selected, present a filtered list to the user; for temperature that would reduce the list to 3 items (°C, °F and K) instead of 3 million different units of measurement.

![image](https://github.com/home-assistant/frontend/assets/10166350/8087d12b-5b24-4fda-b2d7-d65e73e60af1)

**Comment by ildar170975** [comment]
Agree, filtering UoM definitely makes sense.
